src/main/java/chessmaster/game/ChessBoard.java

```diff
@@ -217,7 +221,7 @@ public void executeMove(Move move) throws InvalidMoveException {
             getTileAtCoor(rookStartCoor).setTileEmpty(rookStartCoor);
             getTileAtCoor(rookDestCoor).updateTileChessPiece(rook);
 
-        } else if (move.isRightCastling()) {
+        } else if (move.isRightCastling() && (startCoor.getX() + 3) < SIZE) {
```

Comment:
Can consider making use of the `isOffsetWithinBoard` method in the `Coordinate` class to check for boundary